bug: workflow-tools.js importLocalModule() references src/ paths instead of dist/

## Description

`packages/mcp-server/dist/workflow-tools.js` contains 9 `importLocalModule()` calls that use relative paths pointing to `../../../src/resources/extensions/gsd/...`. Since this file runs from its compiled location in `dist/`, these paths resolve to the `src/` directory where only `.ts` source files exist — not the `.js` files the dynamic `import()` expects.

**Result:** Every MCP tool that calls `importLocalModule()` fails with:
```
Cannot find module '.../src/resources/extensions/gsd/db-writer.js'
```

## Affected modules (all 9)

```
../../../src/resources/extensions/gsd/bootstrap/write-gate.js
../../../src/resources/extensions/gsd/db-writer.js
../../../src/resources/extensions/gsd/doctor.js
../../../src/resources/extensions/gsd/gsd-db.js
../../../src/resources/extensions/gsd/journal.js
../../../src/resources/extensions/gsd/milestone-ids.js
../../../src/resources/extensions/gsd/state.js
../../../src/resources/extensions/gsd/tools/plan-task.js
../../../src/resources/extensions/gsd/tools/workflow-tool-executors.js
```

## Affected tools

`gsd_decision_save`, `gsd_milestone` operations, state writes, journal writes, doctor (via MCP), plan-task — any tool that routes through `importLocalModule()` in `workflow-tools.js`.

## Root cause

`packages/mcp-server/src/workflow-tools.ts` hardcodes relative paths with `src/` prefixes:

```typescript
const { saveDecisionToDb } = await importLocalModule(
  "../../../src/resources/extensions/gsd/db-writer.js"
);
```

TypeScript compiles this to `dist/workflow-tools.js` but string literals pass through unchanged. The `importLocalModule()` function resolves relative to `import.meta.url` (the file's own location), so:

- **Expected resolution:** `packages/mcp-server/dist/` + `../../../dist/resources/extensions/gsd/db-writer.js`
- **Actual resolution:** `packages/mcp-server/dist/` + `../../../src/resources/extensions/gsd/db-writer.js`

## Reproduction

1. Install `gsd-pi` globally: `npm i -g gsd-pi`
2. Run any project with MCP workflow server configured
3. Call `gsd_decision_save` (or any tool using `importLocalModule`)
4. Observe: `Cannot find module '.../src/resources/extensions/gsd/db-writer.js'`

## Workaround

Create symlinks from `dist/*.js` to the `src/` paths:

```bash
BASE="$(npm root -g)/gsd-pi"
for f in \
  resources/extensions/gsd/bootstrap/write-gate.js \
  resources/extensions/gsd/db-writer.js \
  resources/extensions/gsd/doctor.js \
  resources/extensions/gsd/gsd-db.js \
  resources/extensions/gsd/journal.js \
  resources/extensions/gsd/milestone-ids.js \
  resources/extensions/gsd/state.js \
  resources/extensions/gsd/tools/plan-task.js \
  resources/extensions/gsd/tools/workflow-tool-executors.js; do
  ln -sf "$BASE/dist/$f" "$BASE/src/$f"
done
```

## Suggested fix

Change all `importLocalModule()` paths in `packages/mcp-server/src/workflow-tools.ts` from `../../../src/...` to `../../../dist/...`, or introduce a build-time path substitution that rewrites `src/` to `dist/` during compilation.

## Environment

- gsd-pi: global install via npm
- Node: v22.22.0
- OS: Linux (WSL2)

--- a/packages/mcp-server/src/import-candidates.test.ts
+++ b/packages/mcp-server/src/import-candidates.test.ts
@@ -1,0 +1,48 @@
+// GSD-2 — Regression tests for importLocalModule candidate resolution (#3954)
+import { describe, it } from "node:test";
+import assert from "node:assert/strict";
+
+import { _buildImportCandidates } from "./workflow-tools.js";
+
+describe("_buildImportCandidates", () => {
+  it("includes dist/ fallback for src/ paths", () => {
+    const candidates = _buildImportCandidates("../../../src/resources/extensions/gsd/db-writer.js");
+    assert.ok(
+      candidates.some((c) => c.includes("/dist/resources/extensions/gsd/db-writer.js")),
+      "should include dist/ swapped candidate",
+    );
+  });
+
+  it("includes src/ fallback for dist/ paths", () => {
+    const candidates = _buildImportCandidates("../../../dist/resources/extensions/gsd/db-writer.js");
+    assert.ok(
+      candidates.some((c) => c.includes("/src/resources/extensions/gsd/db-writer.js")),
+      "should include src/ swapped candidate",
+    );
+  });
+
+  it("includes .ts variants for .js paths", () => {
+    const candidates = _buildImportCandidates("../../../src/resources/extensions/gsd/db-writer.js");
+    assert.ok(
+      candidates.some((c) => c.endsWith("db-writer.ts") && c.includes("/src/")),
+      "should include .ts variant for original src/ path",
+    );
+    assert.ok(
+      candidates.some((c) => c.endsWith("db-writer.ts") && c.includes("/dist/")),
+      "should include .ts variant for swapped dist/ path",
+    );
+  });
+
+  it("returns original path first", () => {
+    const input = "../../../src/resources/extensions/gsd/db-writer.js";
+    const candidates = _buildImportCandidates(input);
+    assert.equal(candidates[0], input, "first candidate should be the original path");
+  });
+
+  it("handles paths without src/ or dist/ gracefully", () => {
+    const candidates = _buildImportCandidates("./local-module.js");
+    assert.equal(candidates.length, 2, "should have original + .ts variant only");
+    assert.equal(candidates[0], "./local-module.js");
+    assert.equal(candidates[1], "./local-module.ts");
+  });
+});

--- a/packages/mcp-server/src/workflow-tools.ts
+++ b/packages/mcp-server/src/workflow-tools.ts
@@ -337,24 +337,29 @@ function toFileUrl(modulePath: string): string {
   return pathToFileURL(resolve(modulePath)).href;
 }
 
-async function importLocalModule<T>(relativePath: string): Promise<T> {
+/** @internal — exported for testing only */
+export function _buildImportCandidates(relativePath: string): string[] {
   // Build candidate paths: try the given path first, then swap src/<->dist/
   // and try .ts extension. This handles both dev (tsx from src/) and prod
   // (compiled from dist/) execution contexts.
-  const candidates: string[] = [
-    new URL(relativePath, import.meta.url).href,
-  ];
+  const candidates: string[] = [relativePath];
   const swapped = relativePath.includes("/src/")
     ? relativePath.replace("/src/", "/dist/")
     : relativePath.includes("/dist/")
       ? relativePath.replace("/dist/", "/src/")
       : null;
-  if (swapped) candidates.push(new URL(swapped, import.meta.url).href);
+  if (swapped) candidates.push(swapped);
   // Also try .ts variants for dev-mode tsx execution
   if (relativePath.endsWith(".js")) {
-    candidates.push(new URL(relativePath.replace(/\.js$/, ".ts"), import.meta.url).href);
-    if (swapped) candidates.push(new URL(swapped.replace(/\.js$/, ".ts"), import.meta.url).href);
+    candidates.push(relativePath.replace(/\.js$/, ".ts"));
+    if (swapped) candidates.push(swapped.replace(/\.js$/, ".ts"));
   }
+  return candidates;
+}
+
+async function importLocalModule<T>(relativePath: string): Promise<T> {
+  const candidates = _buildImportCandidates(relativePath)
+    .map((p) => new URL(p, import.meta.url).href);
 
   let lastErr: unknown;
   for (const candidate of candidates) {

--- a/packages/mcp-server/src/workflow-tools.ts
+++ b/packages/mcp-server/src/workflow-tools.ts
@@ -326,6 +326,7 @@ function getWriteGateModuleCandidates(): string[] {
 
   candidates.push(
     new URL("../../../src/resources/extensions/gsd/bootstrap/write-gate.js", import.meta.url).href,
+    new URL("../../../dist/resources/extensions/gsd/bootstrap/write-gate.js", import.meta.url).href,
     new URL("../../../src/resources/extensions/gsd/bootstrap/write-gate.ts", import.meta.url).href,
   );
 
@@ -337,7 +338,33 @@ function toFileUrl(modulePath: string): string {
 }
 
 async function importLocalModule<T>(relativePath: string): Promise<T> {
-  return import(new URL(relativePath, import.meta.url).href) as Promise<T>;
+  // Build candidate paths: try the given path first, then swap src/<->dist/
+  // and try .ts extension. This handles both dev (tsx from src/) and prod
+  // (compiled from dist/) execution contexts.
+  const candidates: string[] = [
+    new URL(relativePath, import.meta.url).href,
+  ];
+  const swapped = relativePath.includes("/src/")
+    ? relativePath.replace("/src/", "/dist/")
+    : relativePath.includes("/dist/")
+      ? relativePath.replace("/dist/", "/src/")
+      : null;
+  if (swapped) candidates.push(new URL(swapped, import.meta.url).href);
+  // Also try .ts variants for dev-mode tsx execution
+  if (relativePath.endsWith(".js")) {
+    candidates.push(new URL(relativePath.replace(/\.js$/, ".ts"), import.meta.url).href);
+    if (swapped) candidates.push(new URL(swapped.replace(/\.js$/, ".ts"), import.meta.url).href);
+  }
+
+  let lastErr: unknown;
+  for (const candidate of candidates) {
+    try {
+      return await import(candidate) as T;
+    } catch (err) {
+      lastErr = err;
+    }
+  }
+  throw lastErr;
 }
 
 function getWorkflowExecutorModuleCandidates(env: NodeJS.ProcessEnv = process.env): string[] {
@@ -352,6 +379,7 @@ function getWorkflowExecutorModuleCandidates(env: NodeJS.ProcessEnv = process.en
 
   candidates.push(
     new URL("../../../src/resources/extensions/gsd/tools/workflow-tool-executors.js", import.meta.url).href,
+    new URL("../../../dist/resources/extensions/gsd/tools/workflow-tool-executors.js", import.meta.url).href,
     new URL("../../../src/resources/extensions/gsd/tools/workflow-tool-executors.ts", import.meta.url).href,
   );
 

